edb/server/connpool/pool.py

```diff
@@ -1095,13 +1094,42 @@ def _maybe_tick(self) -> None:
         self._report_snapshot()
         self._capture_snapshot(now=now)
 
+    async def _steal_conn(self, for_block: Block[C]) -> None:
+        # A simplified connection stealing implementation.
+        # First, tries to steal one from the blocks queue unconditionally.
+        for block in self._blocks.values():
+            if block is for_block:
+                continue
+            if (conn := block.try_steal()) is not None:
+                self._log_to_snapshot(
+                    dbname=block.dbname, event='conn-stolen')
+                self._schedule_transfer(block, conn, for_block)
+                self._blocks.move_to_end(block.dbname, last=True)
+                return
+        # If all the blocks are busy, simply wait in the queue to get one.
+        for block in self._blocks.values():
+            if block is for_block:
+                continue
+            if block.count_conns():
```

Comment:
Hm, why should we block until we acquire a connection from a random block? I'm not sure this is right.

Reply:
Ah, this is for the NaivePool. Never mind!